### DISCUSSION TITLE
Include an option for adding contributed commits to Most used languages plugin

### DISCUSSION THREAD

**Discussion by CIAvash** [lowlighter/metrics#319] [Ideas]
If I have not missed anything, I couldn't find an option to include languages I contributed to the repositories of other people(not the whole repositories, just my commits).

**Comment by lowlighter** [comment]
Yes, it's kind a limitation of the `languages` plugin currently 😕

This is because these data are fetched from GitHub Api instead of being computed

To make them more accurate (including for personal repo), one would need to iterate through all commits history to filter out a single user commit and then run github/linguist on it to get stats (the `activity` graphs actually does something similar, but at a smaller scale)

Because of the rate limit it's not possible to do it on the fly (users with thousands of commit will reach the limit quickly), but I can see something working with a cache system.

I know other users whi have been requesting this, I may give it a shoot next week 🙂

**Reply by Nixinova** [reply]
What about just "recent languages" then that can be obtained via the looping through the recent activity that is already being done.

**Comment by lowlighter** [comment]
v3.10 will integrate a `plugin_languages_indepth` option which will perform indepth analysis of languages stats by:
1. cloning repositories you contributed to on the GitHub runner
2. run [github/linguist](https://github.com/github/linguist) on it to guess languages of each file
3. run a `git log --author=login --patch` and for each line compute the number of bytes effectively written

It's still experimental and some huges patches may be skipped but it should still yield nice results (I hope 😅)
When using this mode, it's also possible to display an estimation of lines of code. 

Also, a new `plugin_languages_sections` option will be added to support a new section `recently-used`. This is based mostly on the same algorithm as in `activity` graph and inspired from @Nixinova's suggestion
